sdk/containerregistry/Azure.Containers.ContainerRegistry/api/Azure.Containers.ContainerRegistry.netstandard2.0.cs

```diff
@@ -266,14 +261,36 @@ public partial class DownloadManifestOptions
     {
         public DownloadManifestOptions(string tag = null, string digest = null) { }
```

Comment:
I wonder why we had this type. Service API says that it takes tagOrDigest, but we require user to allocate another object for options just to later on pass any of them.

I'd like to entertain the following idea:
1. Have simple and advanced download overloads
   - `DownloadManifest(string tagOrDigest)`
   - `DownloadManifest(string tagOrDigest, DownloadManifestOptions options)`
2. `DownloadManifestOptions` to only have `MediaType` for the time being

Reply:
Maybe even hide serialization for custom types?
```csharp
DownloadManifestResult<MyManifestListImpl> result = client.downloadManifest(tagOrDigest, new DownloadManifestOptions<MyManifestListImpl>);
MyManifestListImpl manifestList = result.Value; // uses BinaryData.ToObject<T> to serialize
```

Reply:
Yeah, great question :)

Trying to remember back to when we added this type, I think the idea was that we wanted to strongly-type whether we had a tag or a digest, so we could use them differently in the method call.  As well as give us a place to add options if we decided we needed to grow the input parameter list without adding overloads.

Since we're not using them differently, I agree it would make sense to remove the type.  However, do we want to resolve https://github.com/Azure/azure-sdk-for-net/issues/33646 before we decide?

Reply:
> Maybe even hide serialization for custom types?

Offhand, I'm not sure I see the value of adding a convenience overload to support this.  If the caller has a custom type, they're already a somewhat advanced user, and it's not so bad to add an extra line to deserialize to their type?  But feel free to provide more reasons if it feels important ... ?

Reply:
I added the suggestion so we can look at the API.

Reply:
the reason to have 

`MyManifestListImpl manifestList = result.Value;` is to have fewer APIs. Otherwise we'd have `AsOciManifest()`, `AsOciIndext()`, potentially `AsDockerManifest` and so on.

But thinking about it more, it seems users don't know what manifest they will get, so it's impossible to write the first line anyway:
`DownloadManifestResult<MyManifestListImpl> result = client.downloadManifest<MyManifestListImpl>`

Sorry for misleading you :)

Reply:
No worries, thanks @lmolkova!